crates/miden-lib/asm/miden/note.masm

```diff
@@ -114,69 +114,46 @@ end
 #! - RECIPIENT is the commitment to the input note's script, inputs, and the serial number.
 #!
 #! Locals:
-#! - 2: inputs_ptr
-#! - 3: num_inputs (for NoteDetails format)
-#! - 4..7: SERIAL_NUM
-#! - 8..11: SCRIPT_ROOT
-#! - 12..15: INPUTS_COMMITMENT
+#! - 0: inputs_ptr
+#! - 1: num_inputs
 #!
 #! Panics if:
 #! - inputs_ptr is not word-aligned (i.e., is not a multiple of 4).
 #! - num_inputs is greater than 128.
 #!
 #! Invocation: exec
-export.build_recipient.16
-    dup.1 dup.1 loc_store.2 loc_store.3
+export.build_recipient.2
+    dup.1 dup.1 loc_store.0 loc_store.1
     # => [inputs_ptr, num_inputs, SERIAL_NUM, SCRIPT_ROOT]
 
     exec.compute_inputs_commitment
     # => [INPUTS_COMMITMENT, SERIAL_NUM, SCRIPT_ROOT]
 
-    movdnw.2
-    # => [SERIAL_NUM, SCRIPT_ROOT, INPUTS_COMMITMENT]
-
-    dupw.2 dupw.2 dupw.2
-    # => [SERIAL_NUM, SCRIPT_ROOT, INPUTS_COMMITMENT, SERIAL_NUM, SCRIPT_ROOT, INPUTS_COMMITMENT]
-
-    exec.build_recipient_hash
-    # => [RECIPIENT, SERIAL_NUM, SCRIPT_ROOT, INPUTS_COMMITMENT]
-
-    # inserting NoteInputs into AdviceMap
-    dupw.3
-    # => [INPUTS_COMMITMENT, RECIPIENT, SERIAL_NUM, SCRIPT_ROOT, INPUTS_COMMITMENT]
-
-    loc_load.2 loc_load.3 loc_load.2
-    # => [inputs_ptr, num_inputs, inputs_ptr, INPUTS_COMMITMENT, RECIPIENT, SERIAL_NUM, SCRIPT_ROOT, INPUTS_COMMITMENT]
+    loc_load.0 loc_load.1 loc_load.0
+    # => [inputs_ptr, num_inputs, inputs_ptr, INPUTS_COMMITMENT, SERIAL_NUM, SCRIPT_ROOT]
 
     add swap
-    # => [inputs_ptr, end_ptr, INPUTS_COMMITMENT, RECIPIENT, SERIAL_NUM, SCRIPT_ROOT, INPUTS_COMMITMENT]
+    # => [inputs_ptr, end_ptr, INPUTS_COMMITMENT, SERIAL_NUM, SCRIPT_ROOT]
 
     movdn.5 movdn.5
-    # => [INPUTS_COMMITMENT, inputs_ptr, end_ptr, RECIPIENT, SERIAL_NUM, SCRIPT_ROOT, INPUTS_COMMITMENT]
+    # => [INPUTS_COMMITMENT, inputs_ptr, end_ptr, SERIAL_NUM, SCRIPT_ROOT]
 
     adv.insert_mem
-    # => [INPUTS_COMMITMENT, inputs_ptr, end_ptr, RECIPIENT, SERIAL_NUM, SCRIPT_ROOT, INPUTS_COMMITMENT]
-
-    dropw drop drop
-    # => [RECIPIENT, SERIAL_NUM, SCRIPT_ROOT, INPUTS_COMMITMENT]
-
-    movdnw.3
-    # => [SERIAL_NUM, SCRIPT_ROOT, INPUTS_COMMITMENT, RECIPIENT]
+    # => [INPUTS_COMMITMENT, inputs_ptr, end_ptr, SERIAL_NUM, SCRIPT_ROOT]
 
-    # inserting NoteDetails into AdviceMap
-    loc_storew.12 dropw loc_storew.8 dropw loc_storew.4 dropw
-    # => [RECIPIENT]
+    movup.4 drop movup.4 drop
+    # => [INPUTS_COMMITMENT, SERIAL_NUM, SCRIPT_ROOT]
 
-    locaddr.3 dup push.RECIPIENT_DATA_ADVICE_STACK_LENGTH add swap
-    # => [mem_ptr, end_ptr, RECIPIENT]
+    movdnw.2
+    # => [SERIAL_NUM, SCRIPT_ROOT, INPUTS_COMMITMENT]
 
-    movdn.5 movdn.5
-    # => [RECIPIENT, mem_ptr, end_ptr]
+    padw adv.insert_hdword hmerge
+    # => [SN_HASH, SCRIPT_ROOT, INPUTS_COMMITMENT]
```

Comment:
Nit: I know the intention is to keep this concise, but could we still extend `SN_HASH` to `SERIAL_HASH` here and everywhere else to make it more readable?